packages/coin-evm/src/utils/encoder.ts

```diff
@@ -35,6 +35,10 @@ class Structure {
     return this.fields
       .map((field) => {
         const value = properties[field.property];
+        // Use 20B of zeroes to represent empty address
+        if(field.property === 'to'){
+          return (value !== undefined && value !== "") ? field?.encode(value) : field?.encode(0);
+        }
```

Comment:
It is really weird to specify this special case in the encoder.
If the result is not expected, you should change the input data instead.

Reply:
The suggestion has been implemented.